src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java

```diff
@@ -18,21 +18,23 @@
     private ShoppingCartService shoppingCartService;
 
     @Override
-    public Order completeOrder(ShoppingCart shoppingCart) {
-        Order order = new Order(shoppingCart.getUserId());
-        order.setProducts(List.copyOf(shoppingCart.getProducts()));
-        shoppingCartService.clear(shoppingCart);
+    public Order create(Order order) {
         return orderDao.create(order);
     }
 
     @Override
-    public List<Order> getUserOrders(Long userId) {
-        return orderDao.getUserOrders(userId);
+    public Order get(Long id) {
+        return orderDao.get(id).get();
```

Comment:
Method get() in Optional throws NoSuchElementException if value is null. I think it is better to use method orElseThrow() here and throw our exception with message. In future it can be useful to understand where is the problem.
It's only suggestion :)

Reply:
Done